tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -1088,14 +1149,17 @@ SparseGlobalOrderReader::respect_copy_memory_budget(
             condition_.field_names().count(name) != 0)
           return Status::Ok();
 
-        // Get the size for this tile.
+        // Get the size for all tiles.
         uint64_t idx = 0;
         for (; idx < max_cs_idx; idx++) {
           auto rt =
               (ResultTileWithBitmap<uint8_t>*)result_cell_slabs[idx].tile_;
           auto id =
               std::pair<uint64_t, uint64_t>(rt->frag_idx(), rt->tile_idx());
-          if (accounted_tiles.count(id) == 0) {
+          const bool has_tile =
```

Comment:
Can you add a comment to explain what this `has_tile` means and why it needs special treatment?